delivery_carrier_service_level/models/stock_picking.py

```diff
@@ -14,10 +14,16 @@ class StockPicking(models.Model):
         comodel_name="carrier.service.level",
         string="Carrier Service Level",
         domain="[('carrier_id', '=?', carrier_id)]",
+        compute="_compute_carrier_service_level",
+        inverse=lambda x: None,
     )
 
-    @api.onchange("carrier_id")
-    def on_change_carrier_id(self):
-        """Set carrier_service_level_id to False"""
-        for this in self:
-            this.carrier_service_level_id = False
+    def _compute_carrier_service_level(self):
```

Comment:
Compute methods should be above the field definition (convention), More important: they can not be ensure_one, add a `for this in self:` loop. And you will always need to set the value. Set self.carrier_service_level_id to False if no self.carrier_id.